ui/components/table.js

```diff
@@ -68,112 +84,128 @@ export default function Table({
     setSelectedRowIds([])
   }, [pageIndex])
 
-  function toggleAll() {
-    setSelectedRowIds(checkedAll || indeterminate ? [] : data?.map(d => d.id))
-    setCheckedAll(!checkedAll && !indeterminate)
-    setIndeterminate(false)
-  }
-
   return (
     <div className='relative overflow-x-auto rounded-lg border border-gray-200/75'>
       {selectedRowIds.length > 0 && (
         <div className='absolute left-12 flex h-6 items-center py-4 sm:left-16'>
           <button
             type='button'
-            onClick={() => {
-              onDelete(selectedRowIds)
-            }}
+            onClick={() => onDelete()}
             className='rounded-md bg-zinc-50 px-4 py-2 text-2xs font-medium  text-red-500 hover:bg-red-100'
           >
             <div className='flex flex-row items-center'>
               <TrashIcon className='mr-1 mt-px h-3.5 w-3.5' />
-              Remoeve selected
+              Remove selected
             </div>
           </button>
         </div>
       )}
       <table className='w-full text-sm text-gray-600'>
         <thead className='border-b border-gray-200/75 bg-zinc-50/50 text-xs text-gray-500'>
-          {table.getHeaderGroups().map(headerGroup => (
-            <tr key={headerGroup.id}>
-              {allowDelete && data?.length > 0 && (
-                <th scope='col' className='relative w-12 px-6 sm:w-16 sm:px-8'>
-                  <input
-                    type='checkbox'
-                    className='absolute left-4 top-1/2 -mt-2 h-4 w-4 rounded border-gray-300 text-blue-600 focus:ring-blue-500 sm:left-6'
-                    ref={checkbox}
-                    checked={checkedAll}
-                    onChange={toggleAll}
-                  />
-                </th>
-              )}
-              {headerGroup.headers.map(header => (
-                <th
-                  className='w-auto py-2 px-5 text-left font-medium first:max-w-[40%]'
-                  key={header.id}
-                >
-                  {header.isPlaceholder
-                    ? null
-                    : flexRender(
-                        header.column.columnDef.header,
-                        header.getContext()
-                      )}
-                </th>
-              ))}
-            </tr>
-          ))}
+          {table.getHeaderGroups().map(headerGroup => {
+            const disabledBulkDeleteCheckbox =
+              data?.length === 1 && data[0].disabledDeleteCheckbox
```

Comment:
```suggestion
              data?.length === 1 && data?.[0]?.disabledDeleteCheckbox
```

I would recommend renaming this to `enableDeleteCheckbox` since the double negative below can be confusing to read